File: app/src/main/java/mchehab/com/listviewrealm/Person.java
package mchehab.com.listviewrealm;

import org.parceler.Parcel;

import io.realm.PersonRealmProxy;
import io.realm.RealmObject;

/**
 * Created by muhammadchehab on 1/6/18.
 */
@Parcel(implementations = { PersonRealmProxy.class },
        value = Parcel.Serialization.BEAN,
        analyze = { Person.class })
public class Person extends RealmObject{

    protected String firstName;
    protected String lastName;
    protected int age;

    public Person() {
        firstName = lastName = "";
        age = 0;
    }

    public Person(String firstName, String lastName, int age) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void copyPerson(Person person){
        setFirstName(person.getFirstName());
        setLastName(person.getLastName());
        setAge(person.getAge());
    }

}

File: kotlinapp/src/main/java/mchehab/com/kotlinapp/Person.kt
package mchehab.com.kotlinapp

import io.realm.PersonRealmProxy
import io.realm.RealmObject
import io.realm.annotations.RealmClass
import org.parceler.Parcel
import org.parceler.ParcelConstructor
import org.parceler.ParcelProperty

/**
 * Created by muhammadchehab on 1/7/18.
 */
@Parcel(implementations = arrayOf(PersonRealmProxy::class),
        value = Parcel.Serialization.BEAN,
        analyze = arrayOf(Person::class))
@RealmClass
open class Person @ParcelConstructor
constructor(@ParcelProperty("firstName") var firstName: String,
            @ParcelProperty("lastName") var lastName: String,
            @ParcelProperty("age") var age: Int) : RealmObject() {

    constructor() : this("", "", 0)

    fun copy(person: Person){
        firstName = person.firstName
        lastName = person.lastName
        age = person.age
    }
}